### PR TITLE
:seedling: chore: publishes arm64 images with release

### PR DESCRIPTION
**Pull request by faiq**
**What this PR does / why we need it**:
Uses docker build targets for release scripts to reduce repeating logic introduced in #1505. This also makes the release script publish arm64 images for users on m1 macs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
publishes arm64 images with release
```

### CODE REVIEW

**Comment by srm09** [comment]
@faiq I was looking at your changes, and this multi version image release can be done by a simple change in the `hack/release.sh` file which looks something like this below:

```
cluster-api-provider-vsphere on  main [$!?] via 🏎💨 v1.19
at 15:30:48 ➜ git diff
diff --git a/hack/release.sh b/hack/release.sh
index 3e10f165..7a0a0fee 100755
--- a/hack/release.sh
+++ b/hack/release.sh
@@ -102,16 +102,20 @@ function build_images() {
   esac

   # Manager image
-  ARCH=$(go env GOARCH)
-  echo "building ${MANAGER_IMAGE_NAME}:${VERSION} for arch ${ARCH}"
-  docker buildx build --platform linux/"${ARCH}" --output=type=docker --pull \
-    -f Dockerfile \
-    -t "${MANAGER_IMAGE_NAME}":"${VERSION}" \
-    .
-  if [ "${LATEST}" ]; then
-    echo "tagging image ${MANAGER_IMAGE_NAME}:${VERSION} as latest"
-    docker tag "${MANAGER_IMAGE_NAME}":"${VERSION}" "${MANAGER_IMAGE_NAME}":latest
-  fi
+  ARCH=("amd64" "arm64")
+  for arch in "${ARCH[@]}"
+  do
+    local IMG_NAME=${MANAGER_IMAGE_NAME}-${arch}
+    echo "building ${IMG_NAME}:${VERSION} for arch ${arch}"
+    docker buildx build --platform linux/"${arch}" --output=type=docker --pull \
+      -f Dockerfile \
+      -t "${IMG_NAME}":"${VERSION}" \
+      .
+    if [ "${LATEST}" ]; then
+      echo "tagging image ${IMG_NAME}:${VERSION} as latest"
+      docker tag "${IMG_NAME}":"${VERSION}" "${IMG_NAME}":latest
+    fi
+  done
 }

 function logout() {
```

wdyt?

**Comment by faiq** [comment]
thanks @srm09 i agree the changes could be made there, but I felt it was better to reuse the makefile target to follow the principle of DRY code. However since it does touch the makefile as well, I can understand your concerns. 

I'm fine either way, I just need arm64 images :) so if doing it one way will get them for me faster im happy to do it

**Comment by srm09** [comment]
I am of the view that currently the release process does not include the Makefile in any way. We should continue this to unblock this and then work on the refactoring in a follow up issue/PR.

**Comment by faiq** [comment]
sounds good. i'll make the changes

**Comment by faiq** [comment]
@srm09 updated the PR

**Review by srm09 — CHANGES_REQUESTED** [review]
I think the name of the image should be {image-name}-{arch}:{version} instead of the {image-name}:{version}-{arch} as mentioned in the PR.

**Comment by faiq** [comment]
@srm09 updated, sorry I misread the image format, but i think everything should be fine now!

File: hack/release.sh [hunk 1]
```diff
@@ -102,16 +102,19 @@ function build_images() {
...
-    echo "tagging image ${MANAGER_IMAGE_NAME}:${VERSION} as latest"
-    docker tag "${MANAGER_IMAGE_NAME}":"${VERSION}" "${MANAGER_IMAGE_NAME}":latest
-  fi
+  ARCH=("arm64" "amd64")
+  for arch in "${ARCH[@]}"; do
+    echo "building ${MANAGER_IMAGE_NAME}:${VERSION} for arch ${arch}"
+	  docker buildx inspect capv &>/dev/null || docker buildx create --name capv
+    docker buildx build --builder capv --platform linux/"${arch}" --output=type=docker --pull \
+      -f Dockerfile \
+      -t "${MANAGER_IMAGE_NAME}":"${VERSION}-${arch}" \
```
Comment (srm09) [inline hack/release.sh:111]:
```suggestion
      -t "${MANAGER_IMAGE_NAME}-${arch}:"${VERSION}" \
```

File: hack/release.sh [hunk 2]
```diff
@@ -102,16 +102,19 @@ function build_images() {
...
+      .
+    if [ "${LATEST}" ]; then
+      echo "tagging image ${MANAGER_IMAGE_NAME}:${VERSION}-${arch} as latest"
+      docker tag "${MANAGER_IMAGE_NAME}":"${VERSION}-${arch}" "${MANAGER_IMAGE_NAME}:latest-${arch}"
```
Comment (srm09) [inline hack/release.sh:115]:
```suggestion
      echo "tagging image ${MANAGER_IMAGE_NAME}-${arch}:${VERSION} as latest"
      docker tag "${MANAGER_IMAGE_NAME}-${arch}":"${VERSION}" "${MANAGER_IMAGE_NAME}-${arch}:latest"
```

File: hack/release.sh [hunk 3]
```diff
@@ -137,11 +140,26 @@ function push_images() {
   login
 
   # Manager image
-  echo "pushing ${MANAGER_IMAGE_NAME}:${VERSION}"
-  docker push "${MANAGER_IMAGE_NAME}":"${VERSION}"
+  ARCH=("arm64" "amd64")
+  for arch in "${ARCH[@]}"; do
+    echo "pushing ${MANAGER_IMAGE_NAME}:${VERSION}-${arch}"
```
Comment (srm09) [inline hack/release.sh:145]: Same suggestion as above.

File: hack/release.sh [hunk 4]
```diff
@@ -137,11 +140,26 @@ function push_images() {
...
+    --amend "${MANAGER_IMAGE_NAME}:${VERSION}-amd64"
+  docker manifest push "${MANAGER_IMAGE_NAME}:${VERSION}"
   if [ "${LATEST}" ]; then
-    echo "also pushing ${MANAGER_IMAGE_NAME}:${VERSION} as latest"
-    docker push "${MANAGER_IMAGE_NAME}":latest
+    docker manifest create \
+      "${MANAGER_IMAGE_NAME}:latest" \
+      --amend "${MANAGER_IMAGE_NAME}:latest-arm64" \
+      --amend "${MANAGER_IMAGE_NAME}:latest-amd64"
+    docker manifest push "${MANAGER_IMAGE_NAME}:latest"
```
Comment (srm09) [inline hack/release.sh:162]: I am unfamiliar with this, could you elaborate on what this command is doing??

**Review by srm09 — CHANGES_REQUESTED** [review]

File: hack/release.sh (hunk 4 above)
Comment (faiq) [reply hack/release.sh]: yes absolutely, the manifest is saying that the the image at a particular tag has more than one architectures and it points to where those are.

in this case, we are saying the latest image has two architectures and each of those correspond to the existing image tags

**Comment by faiq** [comment]
@srm09 can you please take a look again. the image names are updated with the format

File: hack/release.sh (hunk 2 above)
Comment (faiq) [reply hack/release.sh]: Addressed with common variable

File: hack/release.sh (hunk 1 above)
Comment (faiq) [reply hack/release.sh]: Addressed with common variable

**Comment by srm09** [comment]
/lgtm
/approve

Thanks for submitting this.

**Pull request merged**